app/src/auth/hooks/useWalletConnect.js

```diff
@@ -0,0 +1,134 @@
+import { useMemo, useCallback } from 'react'
+import WalletConnect from '@walletconnect/client'
+import QRCodeModal from '@walletconnect/qrcode-modal'
+import { convertUtf8ToHex } from '@walletconnect/utils'
+import { createChallenge } from '$shared/modules/integrationKey/services'
+import { ChallengeFailedError } from '$shared/errors/Web3'
+import { post } from '$shared/utils/api'
+import routes from '$routes'
+
+const signChallenge = async ({ connector, address }) => {
+    try {
+        const { id, challenge } = await createChallenge(address)
+
+        // encode message (hex)
+        const hexMsg = convertUtf8ToHex(challenge)
+
+        // personal_sign params
+        const msgParams = [hexMsg, address]
+
+        const signature = await connector.signPersonalMessage(msgParams)
+
+        return post({
+            url: routes.auth.external.login(),
+            data: {
+                challenge: {
+                    id,
+                    challenge,
+                },
+                signature,
+                address,
+            },
+        })
+    } catch (error) {
+        console.warn(error)
+        throw new ChallengeFailedError()
+    }
+}
+
+const useWalletConnect = () => {
+    const connect = useCallback(async () => (
+        new Promise((resolve, reject) => {
+            // Create a connector
+            let connector = new WalletConnect({
+                bridge: 'https://bridge.walletconnect.org', // Required
+                qrcodeModal: QRCodeModal,
+            })
+
+            let challengePending = false
+            let connected = false
+            let finalized = false
+
+            const finalize = async ({ token, error, disconnect = true } = {}) => {
+                finalized = true
+
+                if (disconnect && connected) {
+                    await connector.killSession()
+                    connector = undefined
+                    connected = false
+                }
+
+                if (token) {
+                    resolve(token)
+                } else {
+                    reject(error)
+                }
+            }
+
+            const onConnect = (error, payload) => {
+                if (!connected) {
+                    connected = true
+                }
+
+                if (error) {
+                    reject(error)
+                }
+
+                if (!challengePending) {
+                    // Get provided accounts and chainId
+                    const { accounts /* , chainId */ } = payload.params[0]
```

Comment:
`chainId` could be used to verify network but it doesn't seem to matter when logging in